src/main/java/com/ldtteam/structurize/placement/handlers/entity/EntityHandlers.java

```diff
@@ -0,0 +1,226 @@
+import net.minecraftforge.common.ForgeSpawnEggItem;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class containing all Entity Handler implementations.
+ */
+public final class EntityHandlers
+{
+    public static final List<IEntityHandler> handlers = new ArrayList<>();
+    static
+    {
+        handlers.add(new MobEntityHandler());
+        handlers.add(new TextDisplayHandler());
+        handlers.add(new ItemFrameHandler());
+        handlers.add(new HangingEntityHandler());
+        handlers.add(new ArmorStandHandler());
+        handlers.add(new ContainerEntityHandler());
+
+        handlers.add(new DefaultEntityHandler());   // always last
+    }
+
+    /**
+     * Allows for adding new handlers without having to clear the list
+     * in other mods just to override one
+     * @param handler the new handler to add
+     * @param override the class to override if it can be found
+     */
+    public static void add(final IEntityHandler handler, final Class<?> override)
+    {
+        for (int i = 0; i < handlers.size(); i++)
+        {
+            if (override.isInstance(handlers.get(i)))
+            {
+                handlers.set(i, handler);
+                return;
+            }
+        }
+        add(handler);
+    }
+
+    /**
+     * Adds a handler to the start of the handlers list,
+     * effectively overriding existing ones with similar
+     * 'canHandle' functions because this one will evaluate before them
+     * @param handler the new handler to add
+     */
+    public static void add(final IEntityHandler handler)
+    {
+        handlers.add(1, handler);
+    }
+
+    /**
+     * Private constructor to hide implicit one.
+     */
+    private EntityHandlers()
+    {
+        // Intentionally left empty.
+    }
+
+    /**
+     * Handler for all kinds of mob
+     */
+    public static class MobEntityHandler implements IEntityHandler
+    {
+        @Override
+        public boolean canHandle(final Entity entity)
+        {
+            return entity instanceof Mob;
+        }
+
+        @Override
+        public boolean canPlace(final Entity entity, final boolean isCreative, final boolean isFancy)
+        {
+            return isCreative;
+        }
+
+        @Override
+        public List<ItemStack> getRequiredItems(final Entity entity)
+        {
+            final SpawnEggItem egg = ForgeSpawnEggItem.fromEntityType(entity.getType());
+            return egg == null ? List.of() : List.of(new ItemStack(egg));
+        }
+    }
+
+    /**
+     * Handler for text displays.
+     */
+    public static class TextDisplayHandler implements IEntityHandler
+    {
+        @Override
+        public boolean canHandle(final Entity entity)
+        {
+            return entity instanceof Display.TextDisplay;
+        }
+
+        @Override
+        public boolean canPlace(final Entity entity, final boolean isCreative, final boolean isFancy)
+        {
+            return isCreative && !isFancy;

```

Comment:
wonder that all Display types should be placed if `isCreative`, if mcol wants `!isFancy` then it should be there

Reply:
Yeah, we could extend support for those too.